lib/user/lib/src/models/timetable/period.dart

```diff
@@ -82,54 +82,52 @@ class Period {
     return startTimeIsSameOrBefore && endTimeIsSameOrAfter;
   }
 
-  bool isBeforePeriod(Period previous) {
+  bool isBeforePeriod(Period? previous) {
     if (previous == null) return false;
     return startTime.totalMinutes < previous.endTime.totalMinutes;
   }
 
-  bool isAfterPeriod(Period next) {
+  bool isAfterPeriod(Period? next) {
     if (next == null) return false;
     return endTime.totalMinutes > next.startTime.totalMinutes;
   }
 
   bool validate() {
-    if (startTime == null) return false;
-    if (endTime == null) return false;
     if (startTime.totalMinutes > endTime.totalMinutes) return false;
     return true;
   }
 
   @override
-  bool operator ==(other) {
-    return other is Period &&
-        number == other.number &&
-        startTime == other.startTime &&
-        endTime == other.endTime;
-  }
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
 
-  @override
-  int get hashCode {
-    return hash3(number, startTime, endTime);
+    return other is Period &&
+        other.number == number &&
+        other.startTime == startTime &&
+        other.endTime == endTime;
   }
 
   Period copyWith({
-    Time startTime,
-    Time endTime,
+    Time? startTime,
+    Time? endTime,
   }) {
     return Period(
       number: number,
       startTime: startTime ?? this.startTime,
       endTime: endTime ?? this.endTime,
     );
   }
+
+  @override
+  int get hashCode => number.hashCode ^ startTime.hashCode ^ endTime.hashCode;
```

Comment:
```suggestion
  int get hashCode => Object.hash(number, startTime, endTime);
```

btw, since some time you can write this instead